File: rb/features.rb
# SouthParkQuotes SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module SouthParkQuotesFeatures
  def self.make_feature(name)
    case name
    when "base"
      SouthParkQuotesBaseFeature.new
    when "test"
      SouthParkQuotesTestFeature.new
    else
      SouthParkQuotesBaseFeature.new
    end
  end
end

File: py/southparkquotes_sdk/features.py
# SouthParkQuotes SDK feature factory

from southparkquotes_sdk.feature.base_feature import SouthParkQuotesBaseFeature
from southparkquotes_sdk.feature.test_feature import SouthParkQuotesTestFeature


def _make_feature(name):
    features = {
        "base": lambda: SouthParkQuotesBaseFeature(),
        "test": lambda: SouthParkQuotesTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
